common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -544,7 +544,39 @@ protected boolean runAllTasks(long timeoutNanos) {
     }
 
     /**
-     * Invoked before returning from {@link #runAllTasks()} and {@link #runAllTasks(long)}.
+     * Poll all tasks from the task queue and run them via {@link Runnable#run()} method.
+     *
+     * @return {@code true} if at least one task was run.
+     */
+    protected boolean runAllTasks(int maxTasksPerRun) {
```

Comment:
Let's rename it to something like this and call it from `SingleThreadIoEventLoop`.
```suggestion
    protected boolean runAllTasksWithLimit(int maxTasksPerRun) {
```